src/app/forms/custom-sections/tyres/tyres.component.ts

```diff
@@ -21,34 +36,115 @@ export class TyresComponent implements OnInit, OnDestroy, OnChanges {
+    });
 
-    this.form = this.dfs.createForm(template, this.vehicleTechRecord) as CustomFormGroup;
-
-    this._formSubscription = this.form.cleanValueChanges.pipe(debounceTime(400)).subscribe(event => this.formChange.emit(event));
+    this.referenceDataService.loadReferenceData(ReferenceDataResourceType.Tyres);
   }
 
-  ngOnChanges() {
+  ngOnChanges(simpleChanges: SimpleChanges) {
+    const { vehicleTechRecord } = simpleChanges;
+
+    if (vehicleTechRecord.firstChange === false) {
+      const currentAxles = vehicleTechRecord.currentValue.axles;
+      const previousAxles = vehicleTechRecord.previousValue.axles;
+
+      currentAxles.forEach((value: Axle, index: number) => {
+        if (value.tyres?.fitmentCode !== previousAxles[index].tyres.fitmentCode) {
+          this.getTyresRefData(value.tyres!, value.axleNumber!);
+        }
+      });
+    }
+
     this.form?.patchValue(this.vehicleTechRecord, { emitEvent: false });
   }
 
   ngOnDestroy(): void {
     this._formSubscription.unsubscribe();
   }
 
+  get template() {
+    switch (this.vehicleTechRecord.vehicleType) {
+      case VehicleTypes.PSV:
+        return tyresTemplatePsv;
+      case VehicleTypes.HGV:
+        return tyresTemplateHgv;
+      case VehicleTypes.TRL:
+        return tyresTemplateTrl;
+    }
+  }
+
+  get isPsv(): boolean {
+    return this.vehicleTechRecord.vehicleType === VehicleTypes.PSV;
+  }
+
+  get isHgvOrTrl(): boolean {
+    return this.vehicleTechRecord.vehicleType === VehicleTypes.HGV || this.vehicleTechRecord.vehicleType === VehicleTypes.TRL;
+  }
+
   get types(): typeof FormNodeEditTypes {
     return FormNodeEditTypes;
   }
 
+  get widths(): typeof FormNodeWidth {
+    return FormNodeWidth;
+  }
+
+  get speedCategorySymbol(): MultiOptions {
+    return speedCategorySymbolAsOptions();
+  }
+
+  get fitmentCode(): MultiOptions {
+    return fitmentCodeAsOptions();
+  }
+
   get axles(): CustomFormArray {
     return this.form.get(['axles']) as CustomFormArray;
   }
 
+  getAxleTyres(i: number): CustomFormGroup {
+    return this.axles.get([i, 'tyres']) as CustomFormGroup;
+  }
+
+  getTyresRefData(tyre: Tyres, axleNumber: number) {
+    this.isError = false;
+    this.referenceDataService
+      .getByKey$(ReferenceDataResourceType.Tyres, String(tyre.tyreCode))
+      .pipe(take(1))
+      .subscribe({
+        next: data => {
+          try {
+            const refTyre = data as ReferenceDataTyre;
+            const newTyre = new Tyre(tyre);
+            newTyre.tyreSize = refTyre.tyreSize;
+            newTyre.plyRating = refTyre.plyRating;
+            newTyre.dataTrAxles = tyre.fitmentCode === FitmentCode.SINGLE ? Number(refTyre.loadIndexSingleLoad) : Number(refTyre.loadIndexTwinLoad);
+
+            this.addTyreToTechRecord(newTyre, axleNumber);
+          } catch {
+            this.errorMessage = 'Cannot find data of this tyre';
+            this.isError = true;
+            const newTyre = new Tyre(tyre);
```

Comment:
could you declare the `newTyre` variable before your try block so you dont declare it twice (once in try, once in catch)?